akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, StashBuffer }
+import akka.event.Logging
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+import akka.{ actor ⇒ a }
+
+abstract class EventsourcedRequestingRecoveryPermit[Command, Event, State](val context: ActorContext[Any])
+  extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement { same ⇒
+
+  import akka.actor.typed.scaladsl.adapter._
+
+  // has to be lazy, since we want to obtain the persistenceId
+  protected lazy val log = Logging(context.system.toUntyped, this)
+
+  override protected val internalStash = StashBuffer[Any](32) // FIXME what size?
+
+  // --- initialization ---
+  // only once we have a permit, we can become active:
+  requestRecoveryPermit()
+
+  val writerIdentity: WriterIdentity = WriterIdentity.newIdentity()
+
+  // --- end of initialization ---
+
+  // ----------
+
+  def becomeRecovering(): Behavior[Any] = {
+    // FIXME configurable? mostly for "skip snapshot", we don't want to allow partial replays I think
+    val recovery = Recovery()
```

Comment:
Important point I think, to allow skipping snapshots, in case they're corrupted maybe etc?